File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.java
class Solution {
    public int chalkReplacer(int[] chalk, int k) {
        int ret=0;
        int i=0;
        while(true){
            if(chalk[i]>k){
                return i;
            }
            k-=chalk[i];
            i++;
            if(i==chalk.length){
                i=0;
            }
        }
    }
}

File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.cpp
class Solution {
public:
    int chalkReplacer(vector<int>& chalk, int k) {
       long long int sum=0; for(auto it:chalk) sum+=it; 

       sum = (k%sum);

        for(int i=0;i<chalk.size();i++){
            if(sum<chalk[i]) return i;
            sum-=chalk[i];
        }

        return 0;
    }
};